mmtrack/datasets/youtube_vis_dataset.py

```diff
@@ -172,61 +172,64 @@ def evaluate(self, results, metric=['track_segm'], logger=None):
 
         eval_results = dict()
         json_results = self.format_results(results, return_json=True)
-        vis_results = self.convert_vis_fmt()
+        vis_results = self.convert_vis_format()
         track_segm_results = eval_vis(json_results, vis_results, logger)
         eval_results.update(track_segm_results)
 
         return eval_results
 
-    def convert_vis_fmt(self):
-        """Convert the annotation to the format of YouTube-VIS.
+    def convert_vis_format(self):
+        """Convert the annotation to the format of YouTube-VIS. The main
+        difference between the two is the format of 'annotation'. Before
+        modification, it is recorded in the unit of images, and after
+        modification, it is recorded in the unit of instances.
 
         Returns:
             dict: A dict with 3 keys, ``categories``, ``annotations``
                 and ``videos``.
-            - | ``categories`` (dict{list[dict]}): Each list has a dict
-                with 2 keys, ``id`` and ``name``.
-            - | ``videos`` (dict{list[dict]}): Each list has a dict with
-                4 keys of video info, ``id``, ``name``, ``width`` and
-                ``height``.
-            - | ``annotations`` (dict{list[dict]}): Each list has a dict with
-                7 keys of video info, ``category_id``, ``segmentations``,
-                ``bboxes``, ``video_id``, ``areas``, ``id`` and ``iscrowd``.
+            - | ``categories`` (list[dict]): Each dict has 2 keys,
+                ``id`` and ``name``.
+            - | ``videos`` (list[dict]): Each dict has 4 keys of video info,
+                ``id``, ``name``, ``width`` and ``height``.
+            - | ``annotations`` (list[dict]): Each dict has 7 keys of video
+                info, ``category_id``, ``segmentations``, ``bboxes``,
+                ``video_id``, ``areas``, ``id`` and ``iscrowd``.
         """
 
         VIS = defaultdict(list)
         ori_anns = mmcv.load(self.ann_file)
         VIS['categories'] = copy.deepcopy(ori_anns['categories'])
         VIS['videos'] = copy.deepcopy(ori_anns['videos'])
 
-        instance_info = defaultdict(list)
-        frame_id = defaultdict(list)
-        len_video = defaultdict(list)
+        instance_infos = defaultdict(list)
```

Comment:
Deprecate the old code,  we could use the coco api to realize this function as the following:
```python
VIS = defaultdict(list)
the 'categories' and 'videos' has already loaded in this dataset, don't need to reload it.
for ins_id, img_ids in self.instancesToImgs.items():
    instance_infos = []
    for img_id in img_ids:
           for ann in self.coco.imgToAnns[img_id]:
                if ann['instance_id'] == instance_id:
                    add some code to load ann information of this instance
    VIS['annotations'].append(instance_infos)
```